PencilCase/PencilCase/scripts/Pages/PageBase.ts

```diff
@@ -11,7 +11,7 @@
     public isPermanent: boolean;
     public back: () => void = null;
 
-    constructor() {
+    constructor(public parameters?: any) {
```

Comment:
Update page base constructor to make the optional parameters as the base page field. Because we call the initialize method in base class, and it's common to access the parameters in initialize method, but at that time point, the parameters is still not available if it's not page base fields. So, to make it work, move the parameters field to base class.